The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9364525841/job/25777718712) [(plain)](https://github.com/rust-lang/rust/commit/f076dec33622904821b7829ba759da9c610d1451/checks/25777718712/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 exporting to docker image format
#16 sending tarball 29.8s done
#16 DONE 32.5s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---

---- [ui] tests/ui/const-generics/generic_const_exprs/eval-privacy.rs stdout ----
diff of stderr:

7 LL | const fn my_const_fn(val: u8) -> u8 {
8    | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
- error: aborting due to 1 previous error
- error: aborting due to 1 previous error
+ error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
+    |
+    |
+ LL |     type AssocTy = Const<{ my_const_fn(U) }>;
+    |     ^^^^^^^^^^^^ can't leak private type
+ ...
+ LL | const fn my_const_fn(val: u8) -> u8 {
+    | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
+    |
+    = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
+ error: aborting due to 2 previous errors
11 
12 For more information about this error, try `rustc --explain E0446`.
13 
---
To only update this specific test, also pass `--test-args const-generics/generic_const_exprs/eval-privacy.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/const-generics/generic_const_exprs/eval-privacy.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/const-generics/generic_const_exprs/eval-privacy" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/const-generics/generic_const_exprs/eval-privacy/auxiliary"
--- stderr -------------------------------
--- stderr -------------------------------
error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
   |
   |
LL |     type AssocTy = Const<{ my_const_fn(U) }>;
   |     ^^^^^^^^^^^^ can't leak private type
...
LL | const fn my_const_fn(val: u8) -> u8 {
   | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private

error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
   |
   |
LL |     type AssocTy = Const<{ my_const_fn(U) }>;
   |     ^^^^^^^^^^^^ can't leak private type
...
LL | const fn my_const_fn(val: u8) -> u8 {
   | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0446`.
------------------------------------------
------------------------------------------


---- [ui] tests/ui/privacy/where-priv-type.rs stdout ----
diff of stderr:

77 LL | const fn my_const_fn(val: u8) -> u8 {
78    | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
- error: aborting due to 1 previous error; 5 warnings emitted
- error: aborting due to 1 previous error; 5 warnings emitted
+ error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
+    |
+    |
+ LL |     type AssocTy = Const<{ my_const_fn(U) }>;
+    |     ^^^^^^^^^^^^ can't leak private type
+ ...
+ LL | const fn my_const_fn(val: u8) -> u8 {
+    | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
+    |
+    = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
+ error: aborting due to 2 previous errors; 5 warnings emitted
81 
82 For more information about this error, try `rustc --explain E0446`.
83 
---
To only update this specific test, also pass `--test-args privacy/where-priv-type.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/privacy/where-priv-type.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/privacy/where-priv-type" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/privacy/where-priv-type/auxiliary"
--- stderr -------------------------------
--- stderr -------------------------------
warning: type `PrivTy` is more private than the item `S`
   |
LL | pub struct S
LL | pub struct S
   | ^^^^^^^^^^^^ struct `S` is reachable at visibility `pub`
   |
note: but type `PrivTy` is only usable at visibility `pub(crate)`
   |
LL | struct PrivTy;
   | ^^^^^^^^^^^^^
   = note: `#[warn(private_bounds)]` on by default
   = note: `#[warn(private_bounds)]` on by default

warning: type `PrivTy` is more private than the item `E`
   |
LL | pub enum E
LL | pub enum E
   | ^^^^^^^^^^ enum `E` is reachable at visibility `pub`
   |
note: but type `PrivTy` is only usable at visibility `pub(crate)`
   |
LL | struct PrivTy;
   | ^^^^^^^^^^^^^


warning: type `PrivTy` is more private than the item `f`
   |
LL | / pub fn f()
LL | / pub fn f()
LL | | //~^ WARNING type `PrivTy` is more private than the item `f`
LL | |     PrivTy:
LL | |     PrivTy:
   | |___________^ function `f` is reachable at visibility `pub`
   |
note: but type `PrivTy` is only usable at visibility `pub(crate)`
   |
LL | struct PrivTy;
   | ^^^^^^^^^^^^^


warning: type `PrivTy` is more private than the item `S`
   |
LL | / impl S
LL | / impl S
LL | | //~^ WARNING type `PrivTy` is more private than the item `S`
LL | |     PrivTy:
LL | |     PrivTy:
   | |___________^ implementation `S` is reachable at visibility `pub`
   |
note: but type `PrivTy` is only usable at visibility `pub(crate)`
   |
LL | struct PrivTy;
   | ^^^^^^^^^^^^^


warning: type `PrivTy` is more private than the item `S::f`
   |
LL | /     pub fn f()
LL | /     pub fn f()
LL | |     //~^ WARNING type `PrivTy` is more private than the item `S::f`
LL | |         PrivTy:
LL | |         PrivTy:
   | |_______________^ associated function `S::f` is reachable at visibility `pub`
   |
note: but type `PrivTy` is only usable at visibility `pub(crate)`
   |
LL | struct PrivTy;
   | ^^^^^^^^^^^^^


error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
   |
   |
LL |     type AssocTy = Const<{ my_const_fn(U) }>;
   |     ^^^^^^^^^^^^ can't leak private type
...
LL | const fn my_const_fn(val: u8) -> u8 {
   | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private

error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
   |
   |
LL |     type AssocTy = Const<{ my_const_fn(U) }>;
   |     ^^^^^^^^^^^^ can't leak private type
...
LL | const fn my_const_fn(val: u8) -> u8 {
   | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
error: aborting due to 2 previous errors; 5 warnings emitted

For more information about this error, try `rustc --explain E0446`.
------------------------------------------
```

</details>

--- a/tests/ui/const-generics/generic_const_exprs/eval-privacy.rs
+++ b/tests/ui/const-generics/generic_const_exprs/eval-privacy.rs
@@ -15,6 +15,7 @@ where
 {
     type AssocTy = Const<{ my_const_fn(U) }>;
     //~^ ERROR private type
+    //~| ERROR private type
     fn assoc_fn() -> Self::AssocTy {
         Const
     }

--- a/tests/ui/const-generics/generic_const_exprs/eval-privacy.stderr
+++ b/tests/ui/const-generics/generic_const_exprs/eval-privacy.stderr
@@ -7,6 +7,17 @@ LL |     type AssocTy = Const<{ my_const_fn(U) }>;
 LL | const fn my_const_fn(val: u8) -> u8 {
    | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
 
-error: aborting due to 1 previous error
+error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
+  --> $DIR/eval-privacy.rs:16:5
+   |
+LL |     type AssocTy = Const<{ my_const_fn(U) }>;
+   |     ^^^^^^^^^^^^ can't leak private type
+...
+LL | const fn my_const_fn(val: u8) -> u8 {
+   | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
+   |
+   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
+
+error: aborting due to 2 previous errors
 
 For more information about this error, try `rustc --explain E0446`.

--- a/tests/ui/privacy/where-priv-type.rs
+++ b/tests/ui/privacy/where-priv-type.rs
@@ -74,6 +74,7 @@ where
 {
     type AssocTy = Const<{ my_const_fn(U) }>;
     //~^ ERROR private type
+    //~| ERROR private type
     fn assoc_fn() -> Self::AssocTy {
         Const
     }

--- a/tests/ui/privacy/where-priv-type.stderr
+++ b/tests/ui/privacy/where-priv-type.stderr
@@ -77,6 +77,17 @@ LL |     type AssocTy = Const<{ my_const_fn(U) }>;
 LL | const fn my_const_fn(val: u8) -> u8 {
    | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
 
-error: aborting due to 1 previous error; 5 warnings emitted
+error[E0446]: private type `fn(u8) -> u8 {my_const_fn}` in public interface
+  --> $DIR/where-priv-type.rs:75:5
+   |
+LL |     type AssocTy = Const<{ my_const_fn(U) }>;
+   |     ^^^^^^^^^^^^ can't leak private type
+...
+LL | const fn my_const_fn(val: u8) -> u8 {
+   | ----------------------------------- `fn(u8) -> u8 {my_const_fn}` declared as private
+   |
+   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`
+
+error: aborting due to 2 previous errors; 5 warnings emitted
 
 For more information about this error, try `rustc --explain E0446`.

